samples/snap-geometry-edits-with-utility-network-rules/src/main/java/com/esri/arcgismaps/sample/snapgeometryeditswithutilitynetworkrules/components/SnapGeometryEditsWithUtilityNetworkRulesViewModel.kt

```diff
@@ -0,0 +1,514 @@
+            ).onSuccess { identifyResultList ->
+                // As we are using subtype feature layers in this sample the returned features are
+                // contained in the sublayer results
+                val identifiedFeature = identifyResultList.firstOrNull()?.sublayerResults?.firstOrNull()?.geoElements?.firstOrNull()
+                if (identifiedFeature !is ArcGISFeature) {
+                    resetSelections()
+                    return@launch
+                }
+
+                // In this sample we only allow selection of point features. If the identified
+                // feature is null or the feature is not a point feature then reset and return.
+                if (identifiedFeature.featureTable?.geometryType != GeometryType.Point) {
+                    resetSelections()
+                    return@launch
+                } else if (
+                    selectedFeature != null
+                    && identifiedFeature != selectedFeature
+                    && selectedFeature?.featureTable?.layer is FeatureLayer
+                ) {
+                    // If a feature is already selected and the tapped feature is not the selected
+                    // feature then clear the previous selection
+                    (selectedFeature?.featureTable?.layer as FeatureLayer).clearSelection()
+                }
+
+                // Update the selected feature and select it on the layer
+                selectedFeature = identifiedFeature
+                selectedFeature?.let { feature ->
+                    (feature.featureTable?.layer as FeatureLayer).selectFeature(feature)
+
+                    // Create a utility element for the selected feature using the utility network
+                    arcGISMap.utilityNetworks.first().createElementOrNull(feature)?.let { element ->
+                        // Update values for UI based on the selected feature
+                        _assetGroupNameState.value = element.assetGroup.name
+                        _assetTypeNameState.value = element.assetType.name
+                        _isEditButtonEnabled.value = true
+                        setSnapSettings(element.assetType)
+                    } ?: run {
+                        messageDialogVM.showMessageDialog("Error creating UtilityElement")
+                        return@launch
+                    }
+
+                    // Start the editing session once feature is selected.
+                    editFeatureGeometry()
+                }
+            }
+        }
+    }
+
+    /**
+     * Creates [SnapRules] based on the given asset type, synchronizes the snap sources using these
+     * rules, then updates the snap sources list used by the UI.
+     */
+    private suspend fun setSnapSettings(assetType: UtilityAssetType) {
+        // Get the snap rules associated with the asset type
+        val snapRules = SnapRules.create(arcGISMap.utilityNetworks.first(), assetType).getOrThrow()
```

Comment:
Avoid throwing unhandled exceptions in the sample: 
```suggestion
        val snapRules = SnapRules.create(arcGISMap.utilityNetworks.first(), assetType).getOrElse {
            return messageDialogVM.showMessageDialog(it)
        }
```